Autocomplete/predefined-items-and-recent-searches/src/index.js

```diff
@@ -0,0 +1,63 @@
+import algoliasearch from 'algoliasearch/lite';
+import {
+  autocomplete,
+  getAlgoliaHits,
+  reverseHighlightItem,
+} from '@algolia/autocomplete-js';
+import { predefinedItems } from './predefined-items';
+import { recentSearches } from './recent-searches';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+autocomplete({
+  container: '#autocomplete',
+  plugins: [predefinedItems, recentSearches],
+  openOnFocus: true,
+  onStateChange({ state, prevState }) {
+    if (state.query !== prevState.query) {
+      console.log('query:', state.query);
+    }
+  },
+  getSources({ query }) {
+    return [
+      {
+        getInputValue: ({ suggestion }) => suggestion.query,
+        getSuggestions({ query }) {
+          return getAlgoliaHits({
+            searchClient,
+            queries: [
+              {
+                indexName: 'instant_search_demo_query_suggestions',
+                query,
+                params: {
+                  hitsPerPage: 4,
+                  facetFilters: [...recentSearches.data.getFacetFilters()],
```

Comment:
I'm not sure what the need for these facetFilters is?

Reply:
So this is to exclude the items from recent searches in the actual hits.

```js
recentSearches.data.getFacetFilters()
```

returns an array like
```json
[
  'objectID:-aaaaa',
  'objectID:-bbbbb',
  'objectID:-ccccc',
  ...
]
```

Reply:
Ah, gtk, thanks! Will add as a comment in https://github.com/algolia/doc/pull/5033